napari/layers/points/points.py

```diff
@@ -1309,7 +1309,7 @@ def _slice_data(
             else:
                 data = self.data[:, not_disp]
                 distances = np.abs(data - indices[not_disp])
-                matches = np.all(distances < 1e-5, axis=1)
+                matches = np.all(distances <= 0.5, axis=1)
```

Comment:
```suggestion
                width = self.extent.step[not_disp]
                matches = np.all(distances <= width / 2, axis=1)
```

I think you need should take into consideration the scale of the layer as it affects the slider spacing of the dims slider, see

https://github.com/napari/napari/blob/0f5281f641ac9ad468648abdc8b3f30b99ae4b70/napari/layers/base/base.py#L723

Reply:
I was not aware of this logic... Not sure I understand how it works!

Reply:
my bad - i made a mistake here, we havn't multiplied the data by the scale yet, so no need for it to be taken into account